scheme/common/cca/realm/realm_utils.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2024 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+package cca
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/veraison/ccatoken"
+	"github.com/veraison/services/log"
+)
+
+func GetRIM(attr json.RawMessage) (string, error) {
+	var at map[string]interface{}
+	err := json.Unmarshal(attr, &at)
+	if err != nil {
+		return "", fmt.Errorf("unable to get Instance ID: %w", err)
+	}
+	key := "realm-initial-measurement"
+	rim, ok := at[key].(string)
+	if !ok {
+		return "", errors.New("unable to get realm initial measurements")
+	}
+	return rim, nil
+}
+
+func GetRPV(attr json.RawMessage) ([]byte, error) {
+	var at map[string]interface{}
+	err := json.Unmarshal(attr, &at)
+	if err != nil {
+		return nil, err
+	}
+	key := "realm-personalization-value"
+	r, ok := at[key].(string)
+	if !ok {
+		return nil, nil
+	}
```

Comment:
I don't think this is correct.  This test would fail also incase PV is supplied but incorrectly encoded.  And you want to signal this as an error, I think.

There are two conditions to entangled here:

```golang
iv := at["realm-personalization-value]
// key missing => OK
if iv == nil {
    return nil, nil
}
// key not correctly encoded => !OK
v, ok := iv.(string)
if !ok {
    return fmt.Errorf(...)
}
```